src/components/AssetTree/AssetTree.tsx

```diff
@@ -156,26 +138,24 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
     }
     const eventKey = treeNode.props.eventKey;
     const assetId = eventKey ? Number.parseInt(eventKey, 10) : undefined;
+    if (!(assetId && !Number.isNaN(assetId))) {
+      return;
+    }
 
-    if (assetId && !Number.isNaN(assetId)) {
-      const loadedData = await this.cursorApiRequest(assetId);
-      if (loadedData.length > 0) {
-        treeNode.props.dataRef.children = [...loadedData]
-          .sort((a, b) => a.name.localeCompare(b.name))
-          .filter(x => x.parentId && x.parentId === treeNode.props.dataRef.key)
-          .map(x => ({
-            title: AssetTree.getDisplayName(x, this.props.displayName),
-            key: x.id,
-            node: x,
-            isLeaf: loadedData.filter(y => y.parentId === x.id).length <= 0,
-          }));
-
-        this.setState({
-          treeData: [...this.state.treeData],
-        });
-      } else {
-        treeNode.props.dataRef.isLeaf = true;
-      }
+    const cdfAssetChildren = await this.cursorApiRequest(assetId);
+    if (cdfAssetChildren.length > 0) {
+      treeNode.props.dataRef.children = [...cdfAssetChildren]
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .filter(
+          asset =>
+            asset.parentId && asset.parentId === treeNode.props.dataRef.asset.id
+        )
+        .map(asset => AssetTree.convertToNode(asset));
```

Comment:
This looks like a breaking change together with [line 208](https://github.com/cognitedata/gearbox.js/pull/449/files#diff-ebaf1e7f0d00251574a00e040d59e4adR208) as `converToNode` returns an `AssetTreeNode` object incompatible with the `OnSelectAssetTreeParams` interface.

Reply:
@polomani Didn't you fix this?

Reply:
it's here https://github.com/cognitedata/gearbox.js/pull/452